graph/src/components/link_resolver.rs

```diff
@@ -18,6 +19,9 @@ pub type JsonValueStream =
 
 /// Resolves links to subgraph manifests and resources referenced by them.
 pub trait LinkResolver: Send + Sync + 'static + Sized {
+    /// Creates a new link resolver that implements the given timeout.
+    fn with_timeout(&self, timeout: Duration) -> Self;
```

Comment:
Why not take `self` here instead of `&self` and avoid a clone (or leave it to the caller to determine whether a clone is needed or not)

Reply:
Works for me! @lutter PR updated.

Reply:
Looks good to me !